src/CloudIoTCoreMQTTClient.cpp

```diff
@@ -0,0 +1,121 @@
+/******************************************************************************
+ * Copyright 2018 Google
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "CloudIoTCoreMQTTClient.h"
+#include <time.h>
+
+CONFIG_CALLBACK_SIGNATURE = NULL;
+
+void callback(char *topic, uint8_t *payload, unsigned int length) {
+  Serial.print("Message received: ");
+  Serial.println(topic);
+
+  if (configCallback != NULL) {
+    configCallback(payload, length);
+  }
+}
+
+CloudIoTCoreMQTTClient::CloudIoTCoreMQTTClient(CloudIoTCoreDevice &device) {
+  this->device = device;
+  this->mqttClient.setClient(this->client);
+}
+
+CloudIoTCoreMQTTClient::CloudIoTCoreMQTTClient(const char *project_id,
+                                               const char *location,
+                                               const char *registry_id,
+                                               const char *device_id,
+                                               const char *private_key) {
+  this->device = CloudIoTCoreDevice(project_id, location, registry_id,
+                                    device_id, private_key);
+  this->mqttClient.setClient(this->client);
+}
+
+void CloudIoTCoreMQTTClient::connect() {
+  mqttClient.setServer(GOOGLE_APIS_MQTT_HOST, GOOGLE_APIS_MQTT_PORT);
+  mqttClient.setCallback(callback);
+}
+
+void CloudIoTCoreMQTTClient::connectSecure(const char *root_cert) {
+  client.setCACert(root_cert);
+  this->connect();
+}
+
+String CloudIoTCoreMQTTClient::getJWT() {
+  jwt = device.createJWT(time(nullptr));
```

Comment:
Maybe consider caching the JWT if it hasn't yet expired. This is done in the [ESP32 Basic](https://github.com/GoogleCloudPlatform/google-cloud-iot-arduino/blob/master/examples/Esp8266-http/backoff.h) example now.